examples/jsm/nodes/utils/JoinNode.js

```diff
@@ -2,17 +2,18 @@ import TempNode from '../core/Node.js';
 
 class JoinNode extends TempNode {
 
-	constructor( nodes = [] ) {
+	constructor( nodes = [], nodeType = null ) {
 
-		super();
+		super( nodeType );
 
 		this.nodes = nodes;
 
 	}
 
 	getNodeType( builder ) {
 
-		return builder.getTypeFromLength( this.nodes.reduce( ( count, cur ) => count + builder.getTypeLength( cur.getNodeType( builder ) ), 0 ) );
+		const defaultType = builder.getTypeFromLength( this.nodes.reduce( ( count, cur ) => count + builder.getTypeLength( cur.getNodeType( builder ) ), 0 ) );
+		return this.nodeType || defaultType;
```

Comment:
Alternatively, we can change the type based on the componentType of the type of the first node?

Reply:
Yes, I just think it doesn't need to compute the `defaultType` if `this.nodeType` is not null.